File: demos/ios/Demos/Demos/views/ContrastLineChart.m

#import "ContrastLineChart.h"

@implementation ContrastLineChart

- (void)chartRender {
    NSString *jsonPath = [[NSBundle mainBundle] pathForResource:@"Res/mockData_multiLines" ofType:@"json"];
    NSString *jsonData = [NSString stringWithContentsOfFile:jsonPath encoding:NSUTF8StringEncoding error:nil];
    self.chart.canvas(self.canvasView).padding(20, 10, 20, 0.f);
    self.chart.source([F2Utils toJsonArray:jsonData]);
    self.chart.scale(@"value", @{@"nice": @(YES)});
    self.chart.scale(@"date", @{@"tickCount": @(5)});
    self.chart.axis(@"date", @{@"label": @{@"textAlign": @"center"}});
    self.chart.line().position(@"date*value").color(@"type", @[]).fixedSize(2).fixedShape(@"smooth");
    self.chart.legend(@"type", @{@"radius": @(3), @"symbol": @"square"});
    self.chart.tooltip(@{});
    self.chart.render();
}

@end

File: demos/ios/Demos-Swift/Demos-Swift/Views/ContrastLineChart.swift
//
//  ContrastLineChart.swift
//  F2Native
//
//  Created by Westin on 2024-06-09.
//

import Foundation
import F2

class ContrastLineChart: BaseLineChart {
    
    override func chartRender() {
        guard let jsonPath = Bundle.main.path(forResource: "Res/mockData_multiLines", ofType: "json"),
              let jsonDataString = try? String(contentsOfFile: jsonPath),
              let canvasView = self.canvasView,
              let chart = self.chart else {
            return
        }
        let jsonData = F2Utils.toJsonArray(jsonDataString)
        chart.clear()
        chart.canvas()(canvasView).padding()(20, 10, 20, 0)
        chart.source()(jsonData)
        chart.scale()("value", ["nice": true])
        chart.scale()("date", ["tickCount": 5])
        chart.axis()("date", ["label": ["textAlign": "center"]])
        chart.line()().position()("date*value").color()("type", []).fixedSize()(2).fixedShape()("smooth")
        chart.legend()("type", ["radius": 3, "symbol": "square"])
        chart.tooltip()([:])
        chart.animate()(true)
        chart.render()()
    }
}
